x/claim/keeper/claim.go

```diff
@@ -84,40 +97,43 @@ func (k Keeper) GetClaimRecord(ctx sdk.Context, addr string, chain types.Chain)
 }
 
 // GetUserTotalClaimable returns the total claimable amount for an address
-func (k Keeper) GetUserTotalClaimable(ctx sdk.Context, addr string, chain types.Chain) (sdk.Coins, error) {
+func (k Keeper) GetUserTotalClaimable(ctx sdk.Context, addr string, chain types.Chain) (sdk.Coin, error) {
 	claimRecord, err := k.GetClaimRecord(ctx, addr, chain)
 	if err != nil {
-		return sdk.Coins{}, err
+		return sdk.Coin{}, err
 	}
 	if claimRecord.Address == "" {
-		return sdk.Coins{}, nil
+		return sdk.Coin{}, nil
 	}
 
-	totalClaimable := sdk.Coins{}
+	totalClaimable := sdk.NewCoin(claimRecord.InitialClaimableAmount.Denom, sdk.ZeroInt())
 	for action := range types.Action_name {
 		claimableForAction, err := k.GetClaimableAmountForAction(ctx, addr, types.Action(action), chain)
 		if err != nil {
-			return sdk.Coins{}, err
+			return sdk.Coin{}, err
+		}
+		if claimableForAction.IsNil() {
+			continue
 		}
-		totalClaimable = totalClaimable.Add(claimableForAction...)
+		totalClaimable = totalClaimable.AddAmount(claimableForAction.Amount)
 	}
 	return totalClaimable, nil
 }
 
 // GetClaimable returns claimable amount for a specific action done by an address
-func (k Keeper) GetClaimableAmountForAction(ctx sdk.Context, addr string, action types.Action, chain types.Chain) (sdk.Coins, error) {
+func (k Keeper) GetClaimableAmountForAction(ctx sdk.Context, addr string, action types.Action, chain types.Chain) (sdk.Coin, error) {
 	claimRecord, err := k.GetClaimRecord(ctx, addr, chain)
 	if err != nil {
-		return nil, err
+		return sdk.Coin{}, err
 	}
 
 	if claimRecord.Address == "" {
-		return sdk.Coins{}, nil
+		return sdk.Coin{}, nil
 	}
 
 	// if action already completed, nothing is claimable
 	if claimRecord.ActionCompleted[action] {
```

Comment:
I think probably it is a good idea to check  the following
```
if len(claimRecord.ActionCompleted) != 3 {
   return sdk.Coin{}, nil
}
```
If somehow `claimRecord.ActionCompleted` is empty , then , this line will panic